### PR TITLE
Refactor GLTF2Context

### PR DESCRIPTION
**Pull request by jcelerier**
* GLTF2Context became private and a shim class was added thus a massive rename
* https://github.com/KDAB/kuesa/pull/13/files?w=1 can be used to check the changes without whitespace changes (due to clang-format)

### CODE REVIEW

**Comment by jcelerier** [comment]
note: my other PR is based on this one and contains it, maybe I can close it ?

**Comment by mkrus** [comment]
this should probably be rebased on master to update the clang-format file, and run that again...

**Comment by mkrus** [comment]
so all this is included in https://github.com/KDAB/kuesa/pull/16 ?
This can be abandoned then?

File: src/core/gltf2importer/gltf2importer.cpp [hunk 1]
```diff
@@ -327,7 +327,10 @@ void GLTF2Importer::load()
     const QString path = urlToLocalFileOrQrc(m_source);
 
     GLTF2Import::GLTF2Parser parser(m_sceneEntity, m_assignNames);
-    parser.setContext(GLTF2Import::GLTF2ContextPrivate::get(m_context));
+    auto ctx = GLTF2Import::GLTF2ContextPrivate::get(m_context);
```
[the remark this answers is not in the document]
Comment (jcelerier) [reply src/core/gltf2importer/gltf2importer.cpp]: done

**Comment by jcelerier** [comment]
@mkrus the idea (from what I understand, since the decision was likely taken before I joined !) was that making GLTFContext public like this is necessary to pass it between the current GLTF2Importer and a potential GLTF2Exporter in the future. @lemirep what do you think of making the parser public ?

**Pull request merged**